generator/templates/server.go.tmpl

```diff
@@ -147,6 +158,137 @@ type {{ .Name }} struct {
 
 {{- end }}
 
+// FederationServiceCELTypeHelper
+type FederationServiceCELTypeHelper struct {
+	celRegistry    *celtypes.Registry
+	structFieldMap map[string]map[string]*celtypes.FieldType
+	mapMu          sync.Mutex
```

Comment:
Instead of using `sync.Mutex`, it may be better to use `sync.RWMutex`. Then we can use `h.mapMu.RLock()` instead of `h.mapMu.Lock()` when we only read from the map for better concurrency.

Reply:
Certainly, the method you pointed out seems to good way for current implementation 👍 
However, to improve performance, I'm going to stop using map and replace it with switch-case.